uat/integrations/test_integrations.py

```diff
@@ -0,0 +1,72 @@
+# Copyright 2023 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+import logging
+import os
+
+import nbformat
+import pytest
+from nbclient.exceptions import CellExecutionError
+from nbconvert.preprocessors import ExecutePreprocessor
+
+# get the current working directory
+current_directory = os.getcwd()
+
+# change the working directory to the "examples" directory
+EXAMPLES_DIR = "examples"
+examples_directory = os.path.join(current_directory, EXAMPLES_DIR)
+os.chdir(examples_directory)
+
+KATIB_INTEGRATION = {"path": "katib-integration.ipynb", "id": "katib"}
+KFP_INTEGRATION = {"path": "kfp-integration.ipynb", "id": "kfp"}
+MINIO_INTEGRATION = {"path": "minio-integration.ipynb", "id": "minio"}
+MLFLOW_INTEGRATION = {"path": "mlflow-integration.ipynb", "id": "mlflow"}
```

Comment:
Is there a value add that we get by explicitly writing these here, or could we do this programatically?  I'm thinking something that finds all ipynb files in `./examples`, turns them into tests with id of their file path or similar, and then parametrizes the test below.  That way adding a test should be as easy as putting a new notebook in the folder

Reply:
Yep, when I started this I had in mind that we were going to be reusing the `kubeflow-examples` notebooks so I wanted to make sure that we deal with notebook names that we can't control. Given that we now create the tests ourselves though, we should be making sure that they have good names, which should serve as test IDs themselves. I'll look into how we can have a dynamic test discovery process, keeping the PYTHONPATH concerns in mind.

Reply:
Yep, that's exactly how it works now! You just have to add your `ipynb` file anywhere inside the `notebooks` directory.